sdk/state/state.go

```diff
@@ -75,15 +74,12 @@ func NewChannel(c Config) *Channel {
 }
 
 func (c *Channel) NextIterationNumber() int64 {
```

Comment:
This function is a bit subtle to use correctly because in the case we're abandoning a precious unconfirmed payment it returns the wrong value. Can we make this function not exported? Or is it being used in the test now?

Reply:
> This function is a bit subtle to use correctly because in the case we're abandoning a precious unconfirmed payment it returns the wrong value

in which case would that happen? When `latestUnconfirmedPayment` gets reset the function defers to `latestCloseAgreement` for the correct next i, which should always be valid 🤔 .

But ya  it's being used in `state_test.go` currently to make sure the channel iteration numbers are aligned/correct

Reply:
I'm thinking of leaving exported for now, but we can circle back to this if you'd like when you come back online